core/server/master/src/main/java/alluxio/master/block/DefaultBlockMaster.java

```diff
@@ -689,6 +696,54 @@ public BlockInfo getBlockInfo(long blockId) throws BlockInfoException, Unavailab
     return ret;
   }
 
+  private MasterWorkerInfo lookupWorker(WorkerNetAddress workerNetAddress) {
+    for (IndexedSet<MasterWorkerInfo> workers: Arrays.asList(mTempWorkers, mLostWorkers)) {
+      MasterWorkerInfo worker = workers.getFirstByField(ADDRESS_INDEX, workerNetAddress);
+      if (worker != null) {
+        return worker;
+      }
+    }
+    return null;
+  }
+
+  private MasterWorkerInfo lookupWorker(long workerId) {
+    for (IndexedSet<MasterWorkerInfo> workers: Arrays.asList(mTempWorkers, mLostWorkers)) {
+      MasterWorkerInfo worker = workers.getFirstByField(ID_INDEX, workerId);
+      if (worker != null) {
+        return worker;
+      }
+    }
+    return null;
+  }
+
+  private MasterWorkerInfo moveWorker(long workerId) {
```

Comment:
call this `registerWorkerInternal()` or something? I don't think `move` is informative enough.

Could you javadoc this method to describe what it does?

Reply:
`registerWorkerInternal` is good to me